src/main/java/com/cognologix/bankingApplication/services/implementation/CustomerOperationServiceImplementation.java

```diff
@@ -0,0 +1,121 @@
+package com.cognologix.bankingApplication.services.implementation;
+
+import com.cognologix.bankingApplication.dao.BankAccountRepository;
+import com.cognologix.bankingApplication.dao.CustomerRepository;
+import com.cognologix.bankingApplication.dto.Responses.CustomerOperations.BalanceInquiryResponse;
+import com.cognologix.bankingApplication.dto.Responses.CustomerOperations.CreateCustomerResponse;
+import com.cognologix.bankingApplication.dto.Responses.CustomerOperations.CustomerUpdateResponse;
+import com.cognologix.bankingApplication.dto.Responses.CustomerOperations.GetAllCustomerResponse;
+import com.cognologix.bankingApplication.entities.Account;
+import com.cognologix.bankingApplication.entities.Customer;
+import com.cognologix.bankingApplication.exceptions.AccountNotAvailableException;
+import com.cognologix.bankingApplication.exceptions.CustomerAlreadyExistException;
+import com.cognologix.bankingApplication.exceptions.CustomerNotFoundException;
+import com.cognologix.bankingApplication.services.CustomerOperationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.stream.Collectors;
+
+@Service
+public class CustomerOperationServiceImplementation implements CustomerOperationService {
+    @Autowired
+    private BankAccountRepository bankAccountRepository;
+
+    @Autowired
+    private CustomerRepository customerRepository;
+
+    //creating new customer
+    @Override
+    public CreateCustomerResponse createNewCustomer(Customer customer) {
+        //if the customer identity element is already exist then throws exception
+        try {
+            customerRepository.findAll().stream().forEach(customerFromList -> {
+                if (customerFromList.getAdharNumber().equals(customer.getAdharNumber())) {
+                    throw new CustomerAlreadyExistException("Customer is already exist by same adhar number");
+                } else if (customerFromList.getPanCardNumber().equals(customer.getPanCardNumber())) {
+                    throw new CustomerAlreadyExistException("Customer is already exist by same PanCard number");
+                } else if (customerFromList.getEmailId().equals(customer.getEmailId())) {
+                    throw new CustomerAlreadyExistException("Customer is already exist by same email");
+                }
+            });
```

Comment:
make use of findbyAdhaarNUmber  and Pan number

Reply:
Updated by using DAO operations